### PR TITLE
Fixed: Use VertexOuput instead of FragmentInput in prepass.wgsl 

### LINKED ISSUE #9402: WGPU error on switching materials with normal prepass
## Bevy version

v0.11.0

## \[Optional\] Relevant system information

- Rust Version: `1.73.0-nightly (d78bbf4bd 2023-08-03)`
- OS: arch-ish
- Adapter info: `AdapterInfo { name: "Intel(R) HD Graphics 520 (SKL GT2)", vendor: 32902, device: 6422, device_type: IntegratedGpu, driver: "Intel open-source Mesa driver", driver_info: "Mesa 23.0.4", backend: Vulkan }`

## What you did

To reproduce, set up a 3D camera with a `NormalPrepass` component on it. Define a custom material that uses a WGSL fragment. Spawn a scene (I'm not sure what data is important here, but it doesn't work with just a cube; might be UVs), and then set up a system to switch out the `StandardMaterial` for a `CustomMaterial` on any entity that has a `StandardMaterial`. Bevy crashes.

Minimum reproducible [here](https://github.com/johnbchron/bevy_prepass_bug).

## What went wrong

This error occurs:
```
2023-08-09T20:52:12.883651Z ERROR wgpu::backend::direct: Handling wgpu errors as fatal by default
thread 'Compute Task Pool (1)' panicked at /home/jlewis/.cargo/registry/src/index.crates.io-6f17d22bba15001f/wgpu-0.16.3/src/backend/direct.rs:3019:5:
wgpu error: Validation Error

Caused by:
    In Device::create_render_pipeline
      note: label = `prepass_pipeline`
    Error matching ShaderStages(FRAGMENT) shader requirements against the pipeline
    Location[2] is provided by the previous stage output but is not consumed as input by this stage.


note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
Encountered a panic in system `bevy_render::render_resource::pipeline_cache::PipelineCache::process_pipeline_queue_system`!
thread 'Compute Task Pool (0)' panicked at /home/jlewis/.cargo/registry/src/index.crates.io-6f17d22bba15001f/bevy_render-0.11.0/src/pipelined_rendering.rs:135:45:
called `Result::unwrap()` on an `Err` value: RecvError
```

## Additional information

Honestly no clue :(

### PR DESCRIPTION
**Pull request by thebeastxdxd**
Fix Inconsistency between `VertexOutput` and `FragmentInput` in `prepass.wgsl`

# Objective
Fixes #9402 - 
Encountered this bug:
```
2023-08-09T20:52:12.883651Z ERROR wgpu::backend::direct: Handling wgpu errors as fatal by default
thread 'Compute Task Pool (1)' panicked at /home/jlewis/.cargo/registry/src/index.crates.io-6f17d22bba15001f/wgpu-0.16.3/src/backend/direct.rs:3019:5:
wgpu error: Validation Error

Caused by:
    In Device::create_render_pipeline
      note: label = `prepass_pipeline`
    Error matching ShaderStages(FRAGMENT) shader requirements against the pipeline
    Location[2] is provided by the previous stage output but is not consumed as input by this stage.


note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
Encountered a panic in system `bevy_render::render_resource::pipeline_cache::PipelineCache::process_pipeline_queue_system`!
thread 'Compute Task Pool (0)' panicked at /home/jlewis/.cargo/registry/src/index.crates.io-6f17d22bba15001f/bevy_render-0.11.0/src/pipelined_rendering.rs:135:45:
called `Result::unwrap()` on an `Err` value: RecvError
```
The reason was that the `FragmentInput` in the `prepass.wgsl` was missing `@location(2)` which is the `VERTEX_TANGENTS`.

## Solution
- Remove `FragmentInput` in `prepass.wgsl` use `VertexOutput` instead
- Rename `VertexOutput` to `PrepassMeshVertexOutput`
- Rename `clip_position` to `position` in `PrepassMeshVertexOutput`

### CODE REVIEW

**Review by IceSentry — APPROVED** [review]

**Review by JMS55 — APPROVED** [review]

File: crates/bevy_pbr/src/prepass/prepass.wgsl
Comment (superdump) [inline crates/bevy_pbr/src/prepass/prepass.wgsl:162]: It isn’t exactly the same. The meaning of the built-in position is different between vertex output and fragment input and the values will be different. That could be fine, and just mean that the member name should be renamed from clip_position to position or something. Then I see that the tangent is also omitted from the fragment input. Perhaps just because it wasn’t used?

Comment (IceSentry) [reply crates/bevy_pbr/src/prepass/prepass.wgsl:162]: To clarify a bit, we found a bug that was caused because the `FragmentInput` didn't have the vertex tangents defined when the normal prepass is used with a custom shader. After adding them we noticed that it's identical to VertexOutput. The only difference being that `@builtin(position) cliip_position` was never defined in FragmentInput.

Comment (IceSentry) [reply crates/bevy_pbr/src/prepass/prepass.wgsl:162]: We do this a lot with MeshVertexOutput everywhere else in bevy so I think it makes sense to do this here too

Comment (thebeastxdxd) [reply crates/bevy_pbr/src/prepass/prepass.wgsl:162]: but it is true that `clip_position` isn't the correct name for the fragment shader, we can change it to `position` if that would help. I see that in `MeshVertexOutput` it is called `position`.
what do you think  @IceSentry?

Comment (IceSentry) [reply crates/bevy_pbr/src/prepass/prepass.wgsl:162]: Yeah, I'm fine with that part matching `MeshVertexOutput` it makes sense. I'd also suggest to name it `PrepassMeshVertexOutput`. This way it makes it very clear that it's the prepass version.

Comment (thebeastxdxd) [reply crates/bevy_pbr/src/prepass/prepass.wgsl:162]: will do

File: crates/bevy_pbr/src/prepass/prepass.wgsl
Comment (thebeastxdxd) [reply crates/bevy_pbr/src/prepass/prepass.wgsl:144]: fixed in CR

File: crates/bevy_pbr/src/prepass/prepass.wgsl
Comment (thebeastxdxd) [reply crates/bevy_pbr/src/prepass/prepass.wgsl:162]: done

**Comment by superdump** [comment]
I don't want to merge this until after #9258 as that is a big PR and reworking it would be more painful than possibly reworking this one. That also touches this code and changes it a bit so we'll see how things look after that is merged.

**Comment by IceSentry** [comment]
The deferred PR partially fixed the original issue and https://github.com/bevyengine/bevy/pull/10105 will apply the same fix as this PR but in the context of the deferred prepass so I think we can close this. Thank you for the PR @thebeastxdxd

**Pull request closed without merging**
